### PR TITLE
fix: error bar overflow chart area

### PR DESCRIPTION
**Pull request by Yilun-Sun**
## Description

- fix: error bar overflow chart area when set axis allow dataoverflow to true

## Related Issue

#2978 

## Motivation and Context

`Bar` and `Line` are not clip ErrorBarItems correspond to the chart area correctly.

## How Has This Been Tested?

- In local dev env `npm run demo`
- test the CodeSandbox example from #2978

## Screenshots (if appropriate):

CodeSandbox code provided in #2978 :
image

In dev demo env:
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by ckifer — APPROVED** [review]

**Review by ckifer — APPROVED** [review]
I think this looks fine.  Just a question - this happens even when `allowDataOverflow={true}`?

**Comment by Yilun-Sun** [comment]
> I think this looks fine. Just a question - this happens even when `allowDataOverflow={true}`?

As in our doc:
> When domain of the axis is specified and the type of the axis is 'number', if allowDataOverflow is set to be false, the domain will be adjusted when the minimum value of data is smaller than domain[0] or the maximum value of data is greater than domain[1] so that the axis displays all data values. 

> If set to true, graphic elements (line, area, bars) will be clipped to conform to the specified domain.

Change `allowDataOverflow={true}` will clip the chart area to the specific domain as been set, but not for the `ErrorBarItems`, so add clip to `ErrorBarItems` can solve the problem.

**Comment by ckifer** [comment]
Cool, will wait for a review from @nikolasrieble and then merge!

File: src/cartesian/Line.tsx
Comment (nikolasrieble) [inline src/cartesian/Line.tsx:276]: Lets be as specific as possible.
```suggestion
      clipPath: needClip===true ? `url(#clipPath-${clipPathId})` : null,
```

Comment (nikolasrieble) [reply src/cartesian/Line.tsx:276]: I just saw that this line is reimplemented everywhere in the code. So good to go (i.e. for the sake of consistency).

**Pull request merged**
